### PR TITLE
Cleanup code - formatting, CI, etc

### PR DESCRIPTION
**Pull request by lucasdemarchi**

### CODE REVIEW

**Review by jwb99 — COMMENTED** [review]
For the "Auto-format code with black" patch, it would be nice to have a pointer to "black" in the commit message - https://github.com/psf/black

**Review by jwb99 — COMMENTED** [review]
In the "ci: Test installation methods" patch, I don't understand the reference to mkosi in the commit message. It's not critical, but it would be nice to expand the explanation a bit.

**Comment by lucasdemarchi** [comment]
> In the "ci: Test installation methods" patch, I don't understand the reference to mkosi in the commit message. It's not critical, but it would be nice to expand the explanation a bit.

it was meant as "I looked at mkosi .github/workflows/*.yml" and noticed they test multiple installation options. Then I copied and modified the relevant part.

**Comment by lucasdemarchi** [comment]
Reworded the commit messages as requested

File: test/00_basic.bats [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  run --separate-stderr ! git pile whatever-non-existent-subcommand
+  #[[ "${stderr_lines[0]}" = "usage: git-pile"* ]]
+}
+
+@test "help" {
+  run --separate-stderr -0 git pile -h
+  [[ "${lines[0]}" = "usage: git-pile"* ]]
+
+  # use git-pile, as otherwise git itself will try
+  # to open then man page for git-pile, which is not
```
Comment (jwb99) [inline test/00_basic.bats:33]: s/then/the/

File: test/common.bash [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+dump_stderr_console() {
+  printf '## %s\n' "${stderr_lines[@]}" >&3
```
Comment (jwb99) [inline test/common.bash:4]: Disclaimer: I've never used 'bats' before.
My question here is about using file descriptor "3". Essentially, how does this work? Is this related to bats?

Comment (jwb99) [reply test/common.bash:4]: Perfect! Thanks for the explanation.

**Comment by lucasdemarchi** [comment]
Thanks for the review, merged.
